### PR TITLE
Support for partial view templates

### PR DESCRIPTION
**Pull request by justinvdm**
This is needed for #459

### CODE REVIEW

File: go/base/static/js/src/components/views.js [hunk 1]
```diff
@@ -158,10 +158,48 @@
     }
   });
 
+  var TemplateView = Backbone.View.extend({
+    jst: _.template(''),
```
Comment (peterp) [inline go/base/static/js/src/components/views.js]: Could this potentially give you an undesired result in that you might be expecting output and you get none, but it wouldn't be clear why?

File: go/base/static/js/src/components/views.js [hunk 2]
```diff
@@ -158,10 +158,48 @@
...
+
+    partials: {},
+
+    data: {},
+
+    initialize: function(options) {
+      if (options.data) { this.data = options.data; }
+
+      if (options.jst) { this.jst = options.jst; }
+      this.partials = _({}).extend(this.partials, options.partials || {});
```
Comment (peterp) [inline go/base/static/js/src/components/views.js:172]: `({})` ascii art as code. :P

**Comment by peterp** [comment]
:+1:

File: go/base/static/js/src/components/views.js (hunk 2 above)
Comment (justinvdm) [inline go/base/static/js/src/components/views.js:172]: Sigh, was wondering why I never saw anyone else use the `_()` thing with an empty object before.

File: go/base/static/js/src/components/views.js (hunk 1 above)
Comment (justinvdm) [inline go/base/static/js/src/components/views.js]: Good point. Hmm, maybe I should rather have `jst: null`, so we are forced to override it? Thoughts?

Comment (peterp) [inline go/base/static/js/src/components/views.js]: Sounds good to me.

**Pull request merged**